### DISCUSSION TITLE
Tests are currently broken

### DISCUSSION THREAD

**Discussion by tehkillerbee** [EbbLabs/mopidy-tidal#132] [Ideas]
It appears some of the recent changes in the tidalapi has broken some of the tests in mopidy-tidal.

We should probably consider if these tests can be changed, to make them less sensitive to changes made in tidalapi. 

I am a little worried that they may be difficult to maintain due to their complexity. Maybe they can be simplified and or comments can be added to make maintaining the tests easier?

Thoughts @2e0byo ?

**Comment by 2e0byo** [comment]
Now we maintain both it's probably much easier.  There are a few things I can do to make the current approach easier, but really I'd like to add testing into tidalapi, so it has a test mode, returning a test track/artist/etc.  Mocking is horrible and brittle.

That said and I've not looked at what's failed, but web have changed some of the public api recently and tests *ought* to fail in that case.

Might be able to give this a push in two weekend's time.  Shouldn't need too much work.

**Reply by tehkillerbee** [reply]
Yes, I did try to sit down and understand how the tests are written but I think it will take some time to understand. Some of them I am not certain what exactly they are testing. 

My main concern is maintaining the tests, as it should not take too long for a new developer to get familiar with the tests. Otherwise, they risk not being maintained and then the work put into them is gone to waste.

I fixed one of the tests that had failed due to a change in the Visit XX URL. The other ones are slightly more difficult to diagnose and most likely require an expert ;)

**Reply by 2e0byo** [reply]
They're just a bit opaque, particularly as I wrote them to assert then current behaviour before refactoring, rather than with a good understanding of how everything plugged together.  I'll clean things up.
